### PR TITLE
Added enemies. Adding them and dumping their stats

### PR DESCRIPTION
**Pull request by ZelvysK**

### CODE REVIEW

File: MaybeGame/Enemy.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace MaybeGame
+{
+    public class Enemy
+    {
+        EnemyStats enemyStats = new();
+
+        public string EnemyName { get; set; }
+
+        public void DataDump()
```
Comment (ViliusVZ) [inline MaybeGame/Enemy.cs]: Method name

File: MaybeGame/EnemyConfig.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+﻿namespace MaybeGame
+{
+    public static class EnemyConfig
+    {
+        public static Enemy CreateZombie()
+        {
+            var enemy = new Enemy();
+            enemy.StatSet("Zombie");
```
Comment (ViliusVZ) [inline MaybeGame/EnemyConfig.cs]: Pass enum here

File: MaybeGame/EnemyConfig.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            return enemy;
+        }
+
+        public static Enemy CreateSkeleton()
+        {
+            Enemy enemy = new();
+            enemy.StatSet("Skeleton");
```
Comment (ViliusVZ) [inline MaybeGame/EnemyConfig.cs]: And here

File: MaybeGame/EnemyConfig.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            return enemy;
+        }
+        public static Enemy CreateBandit()
+        {
+            Enemy enemy = new();
+            enemy.StatSet("Bandit");
```
Comment (ViliusVZ) [inline MaybeGame/EnemyConfig.cs]: And here

File: MaybeGame/EnemyConfig.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+            return enemy;
+        }
+        public static Enemy CreateTheif()
+        {
+            Enemy enemy = new();
+            enemy.StatSet("Theif");
```
Comment (ViliusVZ) [inline MaybeGame/EnemyConfig.cs]: And here

File: MaybeGame/EnemyConfig.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+            enemy.StatSet("Skeleton");
+            return enemy;
+        }
+        public static Enemy CreateBandit()
+        {
+            Enemy enemy = new();
+            enemy.StatSet("Bandit");
+            return enemy;
+        }
+        public static Enemy CreateTheif()
```
Comment (ViliusVZ) [inline MaybeGame/EnemyConfig.cs]: Theif?

File: MaybeGame/EnemyConfig.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            Enemy enemy = new();
+            enemy.StatSet("Theif");
+            return enemy;
+        }
+        public static Enemy CreateWarrior()
+        {
+            Enemy enemy= new();
+            enemy.StatSet("Warrior");
```
Comment (ViliusVZ) [inline MaybeGame/EnemyConfig.cs]: And here

File: MaybeGame/EnemyConfig.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+            return enemy;
+        }
+        public static Enemy CreateBerserker()
+        {
+            Enemy enemy = new();
+            enemy.StatSet("Berserker");
```
Comment (ViliusVZ) [inline MaybeGame/EnemyConfig.cs]: And here

File: MaybeGame/EnemyConfig.cs [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+            return enemy;
+        }
+        public static Enemy CreateRat()
+        {
+            Enemy enemy = new();
+            enemy.StatSet("Rat");
```
Comment (ViliusVZ) [inline MaybeGame/EnemyConfig.cs]: And here

File: MaybeGame/EnemyConfig.cs [hunk 10]
```diff
@@ -0,0 +1,49 @@
+﻿namespace MaybeGame
+{
+    public static class EnemyConfig
+    {
+        public static Enemy CreateZombie()
+        {
+            var enemy = new Enemy();
```
Comment (ViliusVZ) [inline MaybeGame/EnemyConfig.cs]: If setting stats is all you do during object initialization, pass it to constructor instead

File: MaybeGame/EnemyName.cs [hunk 11]
```diff
@@ -1,6 +1,6 @@
 ﻿namespace MaybeGame
 {
-    public enum EnemyType
+    public enum EnemyName
```
Comment (ViliusVZ) [inline MaybeGame/EnemyName.cs]: EnemyType

File: MaybeGame/EnemyStats.cs [hunk 12]
```diff
@@ -0,0 +1,209 @@
+﻿using System;
+
+namespace MaybeGame
+{
+    public class EnemyStats
```
Comment (ViliusVZ) [inline MaybeGame/EnemyStats.cs]: Stats should be applicable to all kinds of entities. Use same Stats class for Player, Enemy, etc..

File: MaybeGame/EnemyStats.cs [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+        public int Speed { get; set; }
+        public int AttackPower { get; set; }
+
+        public int SetStrength(string enemy)
+        {
+            // Strength is the quality or state of being physically strong.
+
+            var strength = 5;
+
+            switch (enemy)
```
Comment (ViliusVZ) [inline MaybeGame/EnemyStats.cs]: Use switch by EnemyType enum

File: MaybeGame/EnemyStats.cs [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+                case "Berserker":
+                    strength += 7;
+                    break;
+                
+                case "Rat":
+                    strength += 0;
+                    break;
+
+                default:
+                    Console.WriteLine("Error 404: Race not found");
```
Comment (ViliusVZ) [inline MaybeGame/EnemyStats.cs]: 404 is api response type, not applicable to console

File: MaybeGame/EnemyStats.cs [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+        public int SetStamina(string enemy)
+        {
+            // Stamina is the ability to sustain prolonged physical or mental effort.
+
+            var stamina = 5;
+
+            switch (enemy)
+            {
+                case "Zombie":
+                    stamina += 2;
```
Comment (ViliusVZ) [inline MaybeGame/EnemyStats.cs]: Merge some cases with keyword **when**

```suggestion
case when EnemyType.Zombie or EnemyType.Skeleton
                    stamina += 2;
```

File: MaybeGame/EnemyStats.cs [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+                case "Rat":
+                    dexterity += 3;
+                    break;
+            }
+
+            return dexterity;
+        }
+        public int SetHealth(int stamina)
+        {
+            return 50 + (stamina * 3);
```
Comment (ViliusVZ) [inline MaybeGame/EnemyStats.cs]: If this is ALL this method does, consider using expression body **=>**

File: MaybeGame/GameMaster.cs [hunk 17]
```diff
@@ -3,69 +3,92 @@
 
 namespace MaybeGame
 {
-    partial class MainProgram
+    public class GameMaster
```
Comment (ViliusVZ) [inline MaybeGame/GameMaster.cs]:
```suggestion
    public class GameManager
```
GameMaster could be Admin type later on

File: MaybeGame/GameMaster.cs [hunk 18]
```diff
@@ -3,69 +3,92 @@
...
-                        break;
-                }
+            if (playerWantsToQuit) return;
+            else Start();
+        }
 
-                if (playerWantsToQuit) return;
-                else Start();
-            }
+        private void AddNewPlayer()
```
Comment (ViliusVZ) [inline MaybeGame/GameMaster.cs]: Method does too much. Split into two:

1. ParsePlayerData() -> parse player data from console and return it
2. AddNewPlayer() -> uses ParsePlayerData() and adds the new player to list

File: MaybeGame/GameMaster.cs [hunk 19]
```diff
@@ -3,69 +3,92 @@
...
+        {
+            Console.WriteLine("Enter player name");
+            var userName = Console.ReadLine();
+            Console.WriteLine("Enter clan name");
+            var clanName = Console.ReadLine();
+            var player = new Player();
+            player.StatSet(userName, clanName);
```
Comment (ViliusVZ) [inline MaybeGame/GameMaster.cs]: consider moving this to constructor

File: MaybeGame/GameMaster.cs [hunk 20]
```diff
@@ -3,69 +3,92 @@
...
+            _players.Add(player);
+        }
+        private void AddEnemies()
+        {
+            Console.WriteLine("Adding enemies");
+            _enemies.Add(EnemyConfig.CreateZombie());
```
Comment (ViliusVZ) [inline MaybeGame/GameMaster.cs]: Perhaps you could introduce a method **GetDefaultEnemies()** in EnemyConfig and return this list from there instead. Then add it with  _enemies.AddRange() method

File: MaybeGame/Player.cs [hunk 21]
```diff
@@ -4,279 +4,86 @@ namespace MaybeGame
 {
     public class Player
     {
-        //TODO: Make a class to house stats and a class for Experience!!!! Make enum races
+        PlayerStats playerStats = new ();
```
Comment (ViliusVZ) [inline MaybeGame/Player.cs]: private? public?

Comment (ViliusVZ) [reply MaybeGame/Player.cs]:
```suggestion
        private PlayerStats _playerStats = new ();
```

or

```suggestion
        public PlayerStats PlayerStats { get; set; } = new ();
```

File: MaybeGame/Player.cs [hunk 22]
```diff
@@ -4,279 +4,86 @@ namespace MaybeGame
...
-        public int Defence { get; set; }
-        public double DodgeChance { get; set; }
-        public double Speed { get; set; }
-        public int AttackPower { get; set; }
-        public double ExperiencePoints { get; set; }
-        //public double SavedExperiencePoints { get; set; }
-        public int StatPoints { get; set; }
-        public int Level { get; set; }
-
+        public PlayerRaces Race { get; set; }
```
Comment (ViliusVZ) [inline MaybeGame/Player.cs]: Races??? Multiracial character??

File: MaybeGame/Player.cs [hunk 23]
```diff
@@ -4,279 +4,86 @@ namespace MaybeGame
...
-            var selection = int.Parse(Console.ReadLine());
+
+            var success = int.TryParse(Console.ReadLine(), out var selection);
+            if (success == false)
+            {
+                Console.WriteLine("You have chosen an invalid option!");
+                RaceInfo();
+            }
 
             switch (selection)
```
Comment (ViliusVZ) [inline MaybeGame/Player.cs]: Move to

```suggestion
private static PlayerRace ResolvePlayerRace(int race) { }
```

File: MaybeGame/Player.cs [hunk 24]
```diff
@@ -4,279 +4,86 @@ namespace MaybeGame
...
-                Level++;
-                Console.WriteLine("Level up! Congratulations\n" +
-                    $"Your level is now {Level}");
-                ExperiencePoints -= 1;
-            }
-            else
-            {
-                ExperiencePoints += experiencePoints;
-            }
+            return Race;
```
Comment (ViliusVZ) [inline MaybeGame/Player.cs]: Returning Property of a class? Methods should only return scope specific variables

File: MaybeGame/Player.cs [hunk 25]
```diff
@@ -4,279 +4,86 @@ namespace MaybeGame
...
-            Dexterity = SetDexterity(Race);
-            Health = SetHealth(Stamina);
-            Defence = SetDefence(Stamina);
-            DodgeChance = SetDodgeChance(Dexterity);
-            Speed = SetSpeed(Agility);
-            AttackPower = SetAttackPower(Strength);
+            Race = (PlayerRaces) RaceInfo();
+            playerExperience.ExperiencePoints = 0;
+            playerExperience.Level = 1;
+            playerStats.Strength = playerStats.SetStrength(Race.ToString());
```
Comment (ViliusVZ) [inline MaybeGame/Player.cs]: Why .ToString() ? Smells bad here

File: MaybeGame/PlayerRaces.cs [hunk 26]
```diff
@@ -0,0 +1,10 @@
+﻿namespace MaybeGame
+{
+    public enum PlayerRaces
```
Comment (ViliusVZ) [inline MaybeGame/PlayerRaces.cs]:
```suggestion
    public enum PlayerRace
```

**Review by ViliusVZ — CHANGES_REQUESTED** [review]

File: MaybeGame/Enemy.cs (hunk 1 above)
Comment (ZelvysK) [reply MaybeGame/Enemy.cs]: Will name it according to class

File: MaybeGame/EnemyConfig.cs (hunk 2 above)
Comment (ZelvysK) [reply MaybeGame/EnemyConfig.cs]: That would be easier to modify than passing a string

File: MaybeGame/EnemyConfig.cs (hunk 6 above)
Comment (ZelvysK) [reply MaybeGame/EnemyConfig.cs]: English is not my first language

File: MaybeGame/EnemyName.cs (hunk 11 above)
Comment (ZelvysK) [reply MaybeGame/EnemyName.cs]: Yes

File: MaybeGame/EnemyStats.cs (hunk 14 above)
Comment (ZelvysK) [reply MaybeGame/EnemyStats.cs]: This is just a placeholder to let me know if there was some kind of error

File: MaybeGame/EnemyStats.cs (hunk 12 above)
Comment (ZelvysK) [reply MaybeGame/EnemyStats.cs]: Created a StatSettings.cs class

File: MaybeGame/EnemyStats.cs (hunk 13 above)
Comment (ZelvysK) [reply MaybeGame/EnemyStats.cs]: That would be way better

File: MaybeGame/EnemyStats.cs (hunk 16 above)
Comment (ZelvysK) [reply MaybeGame/EnemyStats.cs]: Yes.

File: MaybeGame/GameMaster.cs (hunk 17 above)
Comment (ZelvysK) [reply MaybeGame/GameMaster.cs]: That is true

File: MaybeGame/GameMaster.cs (hunk 20 above)
Comment (ZelvysK) [reply MaybeGame/GameMaster.cs]: Will try to do that

File: MaybeGame/Player.cs (hunk 21 above)
Comment (ZelvysK) [reply MaybeGame/Player.cs]: They are only accessible to player. So private

File: MaybeGame/Player.cs (hunk 25 above)
Comment (ZelvysK) [reply MaybeGame/Player.cs]: Changed the settings to read enum. Do not need to convert to string anymore

File: MaybeGame/Player.cs (hunk 22 above)
Comment (ZelvysK) [reply MaybeGame/Player.cs]: Not completely wrong but they are of different races

File: MaybeGame/PlayerRaces.cs (hunk 26 above)
Comment (ZelvysK) [reply MaybeGame/PlayerRaces.cs]: yes

**Pull request merged**